src/com/facebook/buck/jvm/kotlin/ExternalKotlinc.java

```diff
@@ -0,0 +1,174 @@
+package com.facebook.buck.jvm.kotlin;
+
+import static com.google.common.collect.Iterables.transform;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import com.facebook.buck.io.ProjectFilesystem;
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.rules.BuildRule;
+import com.facebook.buck.rules.RuleKeyObjectSink;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.rules.SourcePathRuleFinder;
+import com.facebook.buck.step.ExecutionContext;
+import com.facebook.buck.util.Console;
+import com.facebook.buck.util.DefaultProcessExecutor;
+import com.facebook.buck.util.ProcessExecutor;
+import com.facebook.buck.util.ProcessExecutorParams;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSortedSet;
+
+public class ExternalKotlinc implements Kotlinc {
+
+  private static final KotlincVersion DEFAULT_VERSION = KotlincVersion.of("unknown version");
```

Comment:
is this the default version on unknown version?
can we document the behavior around versions somewhere? how to set the version, what the default is, what happens if none is set?